safenode/src/protocol/types/wallet/local_store.rs

```diff
@@ -0,0 +1,344 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::{CreatedDbc, DepositWallet, Result, SendClient, SendWallet};
+
+use crate::protocol::types::transfers::Outputs as TransferDetails;
+
+use sn_dbc::{Dbc, DbcId, DbcIdSource, MainKey, PublicAddress, Token};
+
+use async_trait::async_trait;
+use std::collections::{BTreeMap, BTreeSet};
+
+/// A wallet that can send tokens to other addresses.
+pub struct LocalSendWallet<C: SendClient> {
+    client: C,
+    wallet: LocalDepositWallet,
+}
+
+/// A wallet that can only receive tokens.
+pub struct LocalDepositWallet {
+    /// The secret key with which we can access
+    /// all the tokens in the available_dbcs.
+    key: MainKey,
+    /// The current balance of the wallet.
+    balance: Token,
+    /// These are dbcs we've owned, that have been
+    /// spent when sending tokens to other addresses.
+    spent_dbcs: BTreeMap<DbcId, Dbc>,
+    /// These are the dbcs we own that are not yet spent.
+    available_dbcs: BTreeMap<DbcId, Dbc>,
+    /// These are the dbcs we've created by
+    /// sending tokens to other addresses.
+    output_history: Vec<CreatedDbc>,
```

Comment:
what's the diff to the `spent_dbcs` ?
according to the comment, seems they are the same?

Reply:
I don't think the comment describes the same thing?
- `These are the dbcs we've created` -> Creating something means it didn't exist before.
- `These are dbcs we've owned, that have been spent` -> Spending a dbc means it is spent. It's used. Inert, dead.

So, the `output_history` field could probably be renamed to describe that better. But the coments are quite clear I'd say?